sdk/keyvault/azure-security-keyvault-certificates/test/ut/certificate_client_base_test.hpp

```diff
@@ -302,5 +303,18 @@ namespace Azure {
             std::move(result), std::move(secretResponse));
       }
     }
+
+    std::string GetCertificateName(std::string const& name)
+    {
+      std::string returnVal(name);
+      // due to delays in delete and purge we need to have different names for certificates as it
+      // will cause conflicts when trying to create a certificate in the process of
+      // purge/delete
+      if (GetEnv("AZURE_TEST_MODE") == "LIVE")
```

Comment:
this is ugly. No test should ever check or know about LIVE or anything. 

Can we just give unique-name for each test-case?   You can use the name of the test for creating the resource, for instance.

We can remove the delete/purge for all tests where we are not testing that functionality.  The entire Key Vault will be removed as soon as the tests finishes, so we don't need to do create+test+delete+purge for every test case. We just need create+test

Reply:
no, that will ef up the record/playback. if i give random names each time i cannot guarantee the same name at record

Reply:
> if i give random names each time

Don't.  Use the `testName` ( always the same name )